### PR TITLE
helix: new, 0.6.0

### PR DESCRIPTION
**Pull request by WhiredPlanck**
Topic Description
-----------------

Introduce `helix`, a [post-modern](https://helix-editor.com/#:~:text=FAQ-,Post%2Dmodern%3F!,-It%27s%20a%20joke) modal text editor.

Package(s) Affected
-------------------

`helix`

Security Update?
----------------

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

**Comment by WhiredPlanck** [comment]
LGTM on AMD64

图片

File: extra-editors/helix/autobuild/alternative [hunk 1]
```diff
@@ -0,0 +1,3 @@
+alternative /usr/bin/editor /usr/bin/hx 50
+alternative /usr/bin/editor /usr/bin/helix 50
+
```
Comment (MingcongBai) [inline extra-editors/helix/autobuild/alternative]: Trailing blank line.

File: extra-editors/helix/autobuild/beyond [hunk 2]
```diff
@@ -0,0 +1,9 @@
+abinfo "Moving binaries and runtime to /usr/lib ..."
+mkdir -pv "$PKGDIR"/usr/lib/helix
+mv -v "$PKGDIR"/usr/bin/hx "$PKGDIR"/usr/lib/helix
+mv -v "$SRCDIR"/runtime "$PKGDIR"/usr/lib/helix
+
+abinfo "Creating soft link for binaries to /usr/bin ..."
+ln -sv ../lib/helix/hx "$PKGDIR"/usr/bin/hx    # original name
+ln -sv ../lib/helix/hx "$PKGDIR"/usr/bin/helix # alias
+
```
Comment (MingcongBai) [inline extra-editors/helix/autobuild/beyond]: Trailing blank line.

File: extra-editors/helix/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,8 @@
+PKGNAME=helix
+PKGSEC=editors
+PKGDES="A post-modern modal text editor"
+BUILDDEP="rustc llvm"
+
+USECLANG=1
+ABSPLITDBG=0
+
```
Comment (MingcongBai) [inline extra-editors/helix/autobuild/defines]: Trailing blank line.

File: extra-editors/helix/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,8 @@
+PKGNAME=helix
+PKGSEC=editors
+PKGDES="A post-modern modal text editor"
```
Comment (MingcongBai) [inline extra-editors/helix/autobuild/defines]: What does that even mean, it's certainly not helping me decide.

File: extra-editors/helix/autobuild/patches/0001-remove-pre-define-LTO-settings.patch [hunk 5]
```diff
@@ -0,0 +1,9 @@
+diff -aur helix-0.6.0/Cargo.toml helix-0.6.0.aosc/Cargo.toml
```
Comment (MingcongBai) [inline extra-editors/helix/autobuild/patches/0001-remove-pre-define-LTO-settings.patch]: pre-defined *

File: extra-editors/helix/spec [hunk 6]
```diff
@@ -0,0 +1,5 @@
+VER=0.6.0
+SRCS="git::commit=tags/v$VER::https://github.com/helix-editor/helix"
+CHKSUMS="SKIP"
+CHKUPDATE="anitya::id=219661"
+
```
Comment (MingcongBai) [inline extra-editors/helix/spec]: Trailing blank line.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**